src/AzurePipelineParser/Models/Job.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AzurePipelineParser.Models
+{
+    public class Job : IPipelineNode
+    {
+        private List<IPipelineNode> _children = new List<IPipelineNode>();
+
+        public string? Name { get; set; }
+
+        public string? DisplayName { get; set; }
+
+        public IList<string> DependsOn { get; set; } = new List<string>();
+
+        public string? Condition { get; set; }
+
+        // public Strategy? Strategy { get; set; } ToDo: this is a more complex type, should do it later
+
+        public bool ContinueOnError { get; set; }
+
+        public Pool? Pool { get; set; }
+
+        public Workspace? Workspace { get; set; }
+
+        public ContainerReference? Container { get; set; }
+
+        public int? TimeoutInMinutes { get; set; }
+
+        public int? CancelTimeoutInMinutes { get; set; }
+
+        public IList<Variable> Variables { get; set; } = new List<Variable>();
```

Comment:
I think we should have a `VariableNode` that implements `IPipelineNode` and that has Children because essentially in the yaml, variables show as:

```yml
variables:
  <variable definitions> | template reference
```

So that way we would be able to represent the variable tree.